src/modules/Snapper.rb

```diff
@@ -23,14 +23,119 @@
 # Summary:	Snapper settings, input and output functions
 # Authors:	Jiri Suchomel <jsuchome@suse.cz>
 #
-# $Id$
-#
 # Representation of the configuration of snapper.
 # Input and output routines.
+
 require "yast"
 
 module Yast
+
+
+  class Tree
+    attr_accessor :name, :status
+    attr_reader :children
+
+    def initialize(name, parent)
+      @name = name
+      @status = 0
+      @parent = parent
+      @children = []
+    end
+
+    def each()
+      if @parent != nil
+        yield self
+      end
+      @children.each do |subtree|
+        subtree.each do |e|
+          yield e
+        end
+      end
+    end
+
+    def fullname()
+      return @parent ? @parent.fullname() + "/" + @name : @name
+    end
+
+    def created?()
+      return @status & 0x01 != 0
+    end
+
+    def deleted?()
+      return @status & 0x02 != 0
+    end
+
+
+    def icon()
+      if @status == 0
+        return "yast-gray-dot.png"
+      elsif created?
+        return "yast-green-dot.png"
+      elsif deleted?
+        return "yast-red-dot.png"
+      else
+        return "yast-yellow-dot.png"
+      end
+    end
+
+
+    def add(fullname, status)
```

Comment:
I see method names are self-descriptive. But some documentation about methods will be great.